sklearn/utils/_openmp_helpers.pyx

```diff
@@ -59,4 +59,13 @@ cpdef _openmp_effective_n_threads(n_threads=None):
         # OpenMP disabled at build-time => sequential mode
         return 1
 
-    
+
+cdef inline int _openmp_thread_num() nogil:
+    """Return the number of the thread calling this function.
+
+    If scikit-learn is built without OpenMP support, always return 0.
+    """
+    IF SKLEARN_OPENMP_PARALLELISM_ENABLED:
+        return openmp.omp_get_thread_num()
+    ELSE:
+        return 0
```

Comment:
Not used yet, there is no clear motivation for this function (yet). I think we can revert and merge directly into `main` by reverting `_openmp_thread_num` and keeping the rest.

Reply:
Right, I wanted to factor all the small fixtures together but will actually include it with `PairwiseDistancesArgKmin`. :+1: